### PR TITLE
Feature/roc 2513 search for single claim

### PR DESCRIPTION
**Pull request by robertparkinson**
https://tools.hmcts.net/jira/browse/ROC-2521

### CODE REVIEW

File: src/main/features/dashboard/index.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  const requiredRoles = ['solicitor']
+  const unprotectedPaths = []
+  return AuthorizationMiddleware.requestHandler(requiredRoles, accessDeniedCallback, unprotectedPaths)
+}
+
+export class DashboardFeature {
+  enableFor (app: express.Express) {
+    app.all(/^\/(legal\/dashboard.*)$/, dashboardRequestHandler())
+
+    app.all(/^\/legal\/dashboard\/(search|claim-details)$/, DraftMiddleware.requestHandler<DraftDashboard>('dashboardDaft',
```
Comment (timja-kainos) [inline src/main/features/dashboard/index.ts]: any reason that you are only applying this to search and claim-details?

File: src/main/features/dashboard/routes/claim-details.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import { Paths as DashboardPaths } from 'dashboard/paths'
+import ClaimStoreClient from 'claims/claimStoreClient'
+import Claim from 'app/claims/models/claim'
+import ErrorHandling from 'common/errorHandling'
+
+async function renderView (req: express.Request, res: express.Response): Promise<void> {
+
+  try {
+    const claim: Claim = await ClaimStoreClient.retrieveByClaimReference(res.locals.user.dashboardDaft.document.search.reference, res.locals.user.bearerToken)
+    console.log(claim)
```
Comment (timja-kainos) [inline src/main/features/dashboard/routes/claim-details.ts]: update tslint.json file to ban console.logs:
https://github.com/hmcts/cmc-citizen-frontend/blob/master/tslint.json#L5

File: src/main/features/dashboard/routes/claim-details.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import * as express from 'express'
+import { Paths as DashboardPaths } from 'dashboard/paths'
+import ClaimStoreClient from 'claims/claimStoreClient'
+import Claim from 'app/claims/models/claim'
+import ErrorHandling from 'common/errorHandling'
+
+async function renderView (req: express.Request, res: express.Response): Promise<void> {
+
+  try {
```
Comment (timja-kainos) [inline src/main/features/dashboard/routes/claim-details.ts]: try catch not needed with ErrorHandling.apply wrapped around it

File: src/main/features/dashboard/routes/claim-details.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  }
+}
+
+export default express.Router()
+
+  .get(DashboardPaths.claimDetailsPage.uri, ErrorHandling.apply(async (req: express.Request, res: express.Response): Promise<void> => {
+    await renderView(req, res)
+  }))
+
+  .post(DashboardPaths.claimDetailsPage.uri, async (req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> => {
```
Comment (timja-kainos) [inline src/main/features/dashboard/routes/claim-details.ts]: remove this until ready to implement

File: src/main/features/dashboard/views/search.njk [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+{% block content %}
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <form novalidate method="post">
+        {{ csrfProtection(csrf) }}
+
+        {{ linkButton('Issue a new claim', 'start', 'button button-start') }}
+
+        <h2 class="heading-medium" style="margin-bottom: 0">Confirm service</h2>
```
Comment (timja-kainos) [inline src/main/features/dashboard/views/search.njk]: no inline styles, CSP should block that anyway?

File: src/main/features/dashboard/index.ts (hunk 1 above)
Comment (robertparkinson) [reply src/main/features/dashboard/index.ts]: those are the only two pages I've built so far and I'm emulating what was done for the claim journey.

Comment (timja-kainos) [reply src/main/features/dashboard/index.ts]: it probably should just be .+ under the dashboard, adding pages one by one is silly.
you sure the claim journey isn't excluding specific pages?

File: src/main/features/dashboard/routes/claim-details.ts (hunk 2 above)
Comment (robertparkinson) [reply src/main/features/dashboard/routes/claim-details.ts]: done, cheers :)

File: src/main/features/dashboard/views/search.njk (hunk 5 above)
Comment (robertparkinson) [reply src/main/features/dashboard/views/search.njk]: didn't even notice this. was not seeing any errors for it.

File: src/main/app/claims/claimStoreClient.ts [hunk 6]
```diff
@@ -73,8 +73,8 @@ export default class ClaimStoreClient {
           Authorization: `Bearer ${userAuthToken}`
         }
       })
-      .then((claims: object[]) => {
-        return claims.map((claim: object) => new Claim().deserialize(claim))
+      .then((claim) => {
+        return new Claim().deserialize(claim)
```
Comment (timja-kainos) [inline src/main/app/claims/claimStoreClient.ts]: Won't this return a list if you use a solicitor reference?

File: src/main/features/dashboard/views/claim-details.njk [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    <div class="column-two-thirds">
+      <h1 class="heading-large">
+        <span class="form-hint">{{ claimNumber }}</span>
+        {{ claimantVDefendant }}
+      </h1>
+      <h3 class="heading-medium">{{ t('Date Issued') }}
+        <div class="form-label">{{ dateIssued | date }}</div>
+      </h3>
+
+      <h3 class="heading-medium">{{ t('Claimant(s)') }}
```
Comment (yarpo) [inline src/main/features/dashboard/views/claim-details.njk]: Why don't we display `"(s"` only when claimants.length > 1?

File: src/main/features/dashboard/views/search.njk [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+{% block content %}
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <form novalidate method="post">
+        {{ csrfProtection(csrf) }}
+
+        <h1 class="heading-large">{{ t('Manage your claims') }}</h1>
+
+        <h2 class="heading-medium form-group-compound">Find a claim</h2>
```
Comment (yarpo) [inline src/main/features/dashboard/views/search.njk]: translate?

File: src/main/features/dashboard/views/search.njk [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        <p>Once you've found a claim you can:</p>
+
+        <ul class="list-bullet form-group">
+          <li>confirm you've served the claim form on the defendant</li>
```
Comment (yarpo) [inline src/main/features/dashboard/views/search.njk]: translate?

File: src/main/features/dashboard/views/search.njk [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+          <li>download any documents related to the claim</li>
+        </ul>
+
+        {{ searchInput('Search by claim number', 'reference', form, bold=true) }}
+
+      </form>
+    </div>
+    <div class="column-one-third govuk-related-items">
+      <a class="button button-secondary" href="/legal/claim/start">{{ t('Issue a new claim') }}</a>
```
Comment (yarpo) [inline src/main/features/dashboard/views/search.njk]: don't we want to keep this url in some config?

File: src/main/public/stylesheets/components/_form.scss [hunk 11]
```diff
@@ -34,3 +34,34 @@ ul {
...
+  border: 0;
+  box-sizing: content-box;
+  color: $white;
+  cursor: pointer;
+  height: 50px;
+  vertical-align: bottom;
+  width: 50px;
+}
+
+
```
Comment (yarpo) [inline src/main/public/stylesheets/components/_form.scss]: too many empty lines

File: src/main/server.ts [hunk 12]
```diff
@@ -7,6 +7,8 @@ import * as https from 'https'
 
 const port: number = parseInt(process.env.PORT, 10) || 4000
 
+const logger = require('@hmcts/nodejs-logging').getLogger('server')
```
Comment (yarpo) [inline src/main/server.ts]: what's the type of `logger`?

File: src/main/features/dashboard/routes/claim-details.ts [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+import ClaimStoreClient from 'claims/claimStoreClient'
+import Claim from 'app/claims/models/claim'
+import ErrorHandling from 'common/errorHandling'
+
+async function renderView (req: express.Request, res: express.Response): Promise<void> {
+  const claim: Claim = await ClaimStoreClient.retrieveByClaimReference(res.locals.user.dashboardDraft.document.search.reference, res.locals.user.bearerToken)
+  res.render(DashboardPaths.claimDetailsPage.associatedView, {
+    claimNumber: claim.claimNumber,
+    partyStripeValue: claim.claimNumber,
+    claimantVDefendant: claim.claimData.claimants[0].name + ' v ' + claim.claimData.defendants[0].name,
```
Comment (timja-kainos) [inline src/main/features/dashboard/routes/claim-details.ts]: use string template?
```
${claim.claimData.claimants[0].name} v ${claim.claimData.defendants[0].name}
```

I would consider adding a getter to claimData which retrieves the first claimant / defendant
That would make this:
```
${claim.claimData.primaryClaimant.name} v ${claim.claimData.primaryDefendants.name}
```

File: src/main/features/dashboard/views/search.njk [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+{% block content %}
+  <div class="grid-row">
+    <div class="column-two-thirds">
+      <form novalidate method="post">
+        {{ csrfProtection(csrf) }}
+
+        <h1 class="heading-large">{{ t('Manage your claims') }}</h1>
+
+        <h2 class="heading-medium form-group-compound">Find a claim</h2>
+        <p>Once you've found a claim you can:</p>
```
Comment (timja-kainos) [inline src/main/features/dashboard/views/search.njk]: you’ve ?

File: src/main/features/dashboard/views/search.njk (hunk 9 above)
Comment (timja-kainos) [inline src/main/features/dashboard/views/search.njk]: missing translation throughout this page

File: src/main/server.ts [hunk 15]
```diff
@@ -15,10 +17,10 @@ if (app.locals.ENV === 'development' || app.locals.ENV === 'dockertests') {
   }
   const server = https.createServer(serverOptions, app)
   server.listen(port, () => {
-    console.log(`Application started: https://localhost:${port}/legal`)
+    logger.info(`Application started: https://localhost:${port}`)
```
Comment (timja-kainos) [inline src/main/server.ts]: you may want to grab rafal's latest changes in this file. also I don't think you meant to remove `/legal`

File: src/main/app/claims/claimStoreClient.ts (hunk 6 above)
Comment (robertparkinson) [reply src/main/app/claims/claimStoreClient.ts]: we have validation to make sure they only put in a valid claim reference number. Searching by their reference number is not happening yet

File: src/main/features/dashboard/views/claim-details.njk (hunk 7 above)
Comment (robertparkinson) [reply src/main/features/dashboard/views/claim-details.njk]: done

File: src/main/server.ts (hunk 15 above)
Comment (dharmendrak) [reply src/main/server.ts]: we need legal

File: src/main/server.ts [hunk 16]
```diff
@@ -15,10 +17,10 @@ if (app.locals.ENV === 'development' || app.locals.ENV === 'dockertests') {
...
   })
 } else {
   app.listen(port, () => {
-    console.log(`Application started: http://localhost:${port}/legal`)
+    logger.info(`Application started: http://localhost:${port}`)
```
Comment (dharmendrak) [inline src/main/server.ts]: we need legal

File: src/main/features/dashboard/routes/claim-details.ts (hunk 13 above)
Comment (robertparkinson) [reply src/main/features/dashboard/routes/claim-details.ts]: done

File: src/main/features/dashboard/views/search.njk (hunk 10 above)
Comment (robertparkinson) [reply src/main/features/dashboard/views/search.njk]: done

File: src/main/server.ts (hunk 12 above)
Comment (robertparkinson) [reply src/main/server.ts]: no idea.. it's a copy paste job from t1 ;)

File: src/main/features/dashboard/views/search.njk (hunk 14 above)
Comment (robertparkinson) [reply src/main/features/dashboard/views/search.njk]: I just copied the content from the prototype. will check with ronan

File: src/main/features/dashboard/views/search.njk (hunk 9 above)
Comment (robertparkinson) [reply src/main/features/dashboard/views/search.njk]: done

File: src/main/server.ts (hunk 15 above)
Comment (robertparkinson) [reply src/main/server.ts]: I've put back legal but will leave the updating the file to a separate PR as need this merge asap

File: src/main/server.ts (hunk 16 above)
Comment (robertparkinson) [reply src/main/server.ts]: done

File: src/main/server.ts (hunk 12 above)
Comment (yarpo) [reply src/main/server.ts]: They are amateurs ;)

**Pull request merged**